samcli/lib/init/__init__.py

```diff
@@ -90,6 +92,9 @@ def generate_project(
     try:
         LOG.debug("Baking a new template with cookiecutter with all parameters")
         cookiecutter(**params)
+        # Fix gradlew line ending issue caused by Windows git
+        # Putting it after cookiecutter as it will change the line ending
+        _fix_gradlew_line_ending(output_dir)
```

Comment:
So if I'm understanding this, the problem here is that by default `git` on Windows converts LF to CRLF. But when running that inside a (Linux) container with `--use-container`, the CRLF endings mess up the parsing?

Doesn't this mean `gradlew` is essentially unreadable on Windows afterwards, as the line endings are LF?

Reply:
gradlew is a shell script and Windows can read both CRLF and LF.

Reply:
Won't it show up weird in Notepad for example?

I might not have full context here, but it feels this is fixing a symptom and not the root cause. Looking at the [Gradle docs](https://docs.gradle.org/current/userguide/gradle_wrapper.html), it seems there's supposed to be a `gradlew.bat` as well?

Reply:
Actually never mind if it's a shell script it makes sense. Can you add a comment explaining that? Since it's what makes this essential to understand.

Reply:
If Windows can handle LF, can we disable `core.autocrlf`?

Reply:
> If Windows can handle LF, can we disable `core.autocrlf`?

It's a global user preference, we shouldn't be changing that.

Reply:
Can't it be disabled for the pull only?

Reply:
We can add specific file .gitattributes in sam-app-template repo, but cookiecutter will still mess up the line endings as the processing is done after git. Related PR: https://github.com/cookiecutter/cookiecutter/pull/1407.